lib/Compiler.js

```diff
@@ -144,7 +143,7 @@ class Compiler extends Tapable {
 			if (err) return callback(err);
 
 			if (this.hooks.shouldEmit.call(compilation) === false) {
-				const stats = new Stats(compilation);
+				const stats = compilation.getStats();
```

Comment:
If `new Stats` can no longer be used, it would be a breaking change.

Should we go this way. (Would mean waiting until webpack 5)

Or can be solve this problem in a non-breaking way.

Reply:
`new Stats`  can be used, but i thought it would be better if we just called `compilation.getStats()` and after this call, we can refer current `stats` through `compilation.stats` and modify it accordingly, but we can surely go through non-breaking way if you suggest. Will update it.